lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -602,9 +602,56 @@ func (d *ceph) UpdateVolume(vol Volume, changedConfig map[string]string) error {
 	return nil
 }
 
+type cephDuInfo struct {
+	Images []cephDuLine `json:"images"`
+}
+
+type cephDuLine struct {
+	Name            string `json:"name"`
+	Snapshot        string `json:"snapshot"`
+	ProvisionedSize int64  `json:"provisioned_size"`
+	UsedSize        int64  `json:"used_size"`
+}
```

Comment:
As not used outside of the function, you should put those in the function directly, see `GetResources` in driver_ceph.go for an example.

Reply:
Didn't even know you could define structs in functions, updated.

Reply:
You can define constants too :)